sdk/eventhub/azure-eventhub/azure/eventhub/_client_base.py

```diff
@@ -61,7 +64,17 @@ def _parse_conn_str(conn_str, kwargs):
             shared_access_key = value
         elif key.lower() == "entitypath":
             entity_path = value
-    if not all([endpoint, shared_access_key_name, shared_access_key]):
+        elif key.lower() == "sharedaccesssignature":
+            shared_access_signature = value
+            try:
+                # Expiry can be stored in the "se=<timestamp>" clause of the token. ('&'-separated key-value pairs)
+                # type: ignore
+                shared_access_signature_expiry = int(shared_access_signature.split('se=')[1].split('&')[0])
+            except (IndexError, TypeError, ValueError): # Fallback since technically expiry is optional.
+                # An arbitrary, absurdly large number, since you can't renew.
+                shared_access_signature_expiry = int(time.time() * 2)
+            print(shared_access_signature_expiry)
+    if not (all([endpoint, shared_access_key_name, shared_access_key]) or all([endpoint, shared_access_signature])):
```

Comment:
Nitpicking. tuple is better than a list for this case?

Reply:
I have no strong feelings.  Adjusting++